moirae/estimators/online/utils/model.py

```diff
@@ -23,17 +24,39 @@ def convert_vals_model_to_filter(
         a corresponding MultivariateRandomDistribution (either Gaussian or Delta)
     """
     if uncertainty_matrix is None:
-        return DeltaDistribution(mean=model_quantities.to_numpy()[0, :])
-    return MultivariateGaussian(mean=model_quantities.to_numpy()[0, :], covariance=uncertainty_matrix)
+        return DeltaDistribution(mean=model_quantities.to_numpy().flatten())
+    return MultivariateGaussian(mean=model_quantities.to_numpy().flatten(), covariance=uncertainty_matrix)
 
 
-# TODO (wardlt): Implement the "ASOHOnly" interface needed by the Dual Estimator by making it such that
-#  the `predict_output` function first estimates how the transients will update for each set of ASOH,
-#  then uses the updated transient states and ASOH to determine the outptus
+def convert_numpy_to_model(filter_array: np.ndarray,
+                           template: Union[GeneralContainer, HealthVariable],
+                           names: Optional[Tuple[str, ...]] = None) -> Union[GeneralContainer, HealthVariable]:
+    """
+    Function to convert numpy arrays to model-related object given a template of said object: either a
+    :class:`~moirae.model.base.GeneralContainer` or a :class:`~moirae.model.base.HealthVariable`.
+
+    Args:
+        filter_array: array of numerical values to be converted
+        template: template object to be used; original object will not be modified
+        names: in case of a `HealthVariable`, can update specific names
+
+    Returns:
+        corresponding model-related object given by template with numerical values from the array
+    """
+    model_related_object = template.model_copy(deep=True)

```

Comment:
This method seems like it would make more sense as a method for the `GeneralContainer` and the `HealthVariable` class for two reasons:

1. The logic is very specific to the type of the input. There is only one line in common between the two types, and one of three arguments is not applicable to GeneralContainer.
2. The logic is not very specific to the use of CellModels in Filters. There are many cases where someone wants a new copy of a container but with updated values?

Could you add this method to those two classes? Maybe call it "make_copy(self, new_values)"

Reply:
Yep, taking care of it!